react/src/components/PrometheusMetricGraph.tsx

```diff
@@ -0,0 +1,255 @@
+import BAICard from './BAICard';
+import {
+  PrometheusMetricGraphQuery,
+  PrometheusMetricGraphQuery$data,
+} from './__generated__/PrometheusMetricGraphQuery.graphql';
+import { Empty, theme } from 'antd';
+import { createStyles } from 'antd-style';
+import graphql from 'babel-plugin-relay/macro';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+import { useLazyLoadQuery } from 'react-relay';
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+  ReferenceLine,
+} from 'recharts';
+
+const useStyle = createStyles(({ css, token }) => ({
+  recharts: css`
+    .recharts-label {
+      fill: ${token.colorTextDescription};
+    }
+    .recharts-cartesian-axis-line {
+      stroke: ${token.colorBorder};
+    }
+    .recharts-cartesian-axis-tick-line {
+      stroke: ${token.colorBorder};
+    }
+    .recharts-cartesian-axis-tick-value {
+      fill: ${token.colorTextDescription};
+    }
+  `,
+}));
+
+type MetricData = NonNullable<
+  NonNullable<
+    PrometheusMetricGraphQuery$data['user_utilization_metric']
+  >['metrics']
+>;
+
+interface PrometheusMetricGraphProps {
+  startDate: string;
+  endDate: string;
+  metricName: string;
+  fetchKey: string;
+}
+
+const PrometheusMetricGraph: React.FC<PrometheusMetricGraphProps> = ({
+  startDate,
+  endDate,
+  metricName,
+  fetchKey,
+}) => {
+  const { token } = theme.useToken();
+  const { styles } = useStyle();
+  const userInfo = useCurrentUserInfo();
+  const dayDiff = dayjs(Number(endDate) * 1000).diff(
+    dayjs(Number(startDate) * 1000),
+    'day',
+  );
+
+  const { user_utilization_metric } =
+    useLazyLoadQuery<PrometheusMetricGraphQuery>(
+      graphql`
+        query PrometheusMetricGraphQuery(
+          $user_id: UUID!
+          $props: UserUtilizationMetricQueryInput!
+        ) {
+          user_utilization_metric(user_id: $user_id, props: $props) {
+            user_id
+            metrics {
+              metric_name
+              value_type
+              values {
+                timestamp
+                value
+              }
+              max_value
+              avg_value
+            }
+          }
+        }
+      `,
+      {
+        user_id: userInfo[0]?.uuid ?? '',
+        props: {
+          metric_name: metricName,
+          start: startDate,
+          end: endDate,
+          step: dayDiff < 7 ? '5m' : dayDiff < 30 ? '1h' : '1d',
+        },
+      },
+      {
+        fetchPolicy: 'store-and-network',
+        fetchKey,
+      },
+    );
+
+  const metricData = getMetricData(
+    user_utilization_metric?.metrics ?? [],
+    startDate,
+    endDate,
+    dayDiff < 7 ? '5m' : dayDiff < 30 ? '1h' : '1d',
+  );
+
+  return (
+    <BAICard
+      title={_.startCase(metricName.replaceAll('_', ' '))}
+      type="inner"
+      styles={{
+        body: {
+          padding: `${token.marginMD}px ${token.marginMD}px ${token.marginXS}px ${token.marginMD}px`,
```

Comment:
Hmm.. why did you set different margin for only bottom? Wouldn't it be better to set it to the same one?

Reply:
The graph info part of the rechart has a default margin set. If you apply the same to the bottom margin, the bottom part of the card will be set too large.

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/lSyr8xXz1wdXALkJKzVx/f80ecdb7-9a0d-473c-abcc-0529e0f2cf49.png)

Reply:
Hmm... I don't think so. Let's listen to other members' opinion.